nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-pubsub-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/KafkaPublisher.java

```diff
@@ -100,15 +107,31 @@ KafkaPublisherResult publish(PublishingContext publishingContext) {
 
         byte[] messageBytes;
         int tokenCounter = 0;
-        for (; (messageBytes = streamTokenizer.nextToken()) != null; tokenCounter++) {
+        boolean continueSending = true;
+        KafkaPublisherResult result = null;
+        for (; (messageBytes = streamTokenizer.nextToken()) != null && continueSending; tokenCounter++) {
```

Comment:
Would recommend switching the orders of the conditions here, as mentioned above.